code/gateway/dataaccess.q

```diff
@@ -0,0 +1,107 @@
+\d .dataaccess
+
+forceservers:0b
+
+timebarmap:`nanosecond`timespan`microsecond`second`minute`hour`day!1 1 1000 1000000000 60000000000 3600000000000 86400000000000;
+
+// function to convert sorting
+go:{if[`asc=x[0];:(xasc;x[1])];:(xdesc;x[1])};
+
+// Full generality dataaccess function in the gateway
+getdata:{[o]
+    // Input checking in the gateway
+    reqno:.requests.initlogger[o];
```

Comment:
Are we recording requests?

Reply:
Yeah the reqno is essentially a ticket number for which entry in the logging table the query relates. If logging is turned off the 'ticketnumber' is set to -1 and the requests aren't logged.